compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1703,6 +1703,13 @@ fn add_post_link_args(cmd: &mut dyn Linker, sess: &Session, flavor: LinkerFlavor
 /// that are necessary for the linking. They are only present in symbol table but not actually
 /// used in any sections, so the linker will therefore pick relevant rlibs for linking, but
 /// unused `#[no_mangle]` or `#[used]` can still be discard by GC sections.
+///
+/// There's a few internal crates in the standard library (aka libcore and
+/// libstd) which actually have a circular dependence upon one another. This
+/// currently arises through "weak lang items" where libcore requires things
+/// like `rust_begin_unwind` but libstd ends up defining it. To get this
+/// circular dependence to work correctly we declare some of these things
+/// in this synthetic object.
```

Comment:
These comments should be moved as well

Reply:
Currently, purpose of creating synthetic object file completely described here. This comment also describing purpose of this file. Shouldn't it be in the same place?

Reply:
That's reasonable. However then please put some comments on the code that adds these symbols to refer to this comment, so the reader of the code can follow through and understand why it's being done.